bodhi/tests/server/services/test_updates.py

```diff
@@ -5194,3 +5194,87 @@ def test_waive_test_results_2_unsatisfied_requirements(
                 u'user'
             ]
         )
+
+    @mock.patch.dict(config, [('test_gating.required', True)])
+    @mock.patch('bodhi.server.models.waiverdb_api_post')
+    @mock.patch('bodhi.server.models.greenwave_api_post')
+    def test_waive_test_results_1_of_2_unsatisfied_requirements(
```

Comment:
I suggest one more test that tries to waive 2 test results, to make sure the schema can handle lists.